json-type.mdx

```diff
@@ -0,0 +1,308 @@
+
+We plan on extending Xata support for JSON even further in the future, but the current capabilities are already very powerful and will solve most use cases.
+Now we are going to provide a few examples on how you can start using JSON documents in Xata today.
+
+Let's think a bit about a simple data model for an online shop. Suppose we have a _Products_ table.
+
+![Products table](/images/json-type/products.png)
+
+## Creating a JSON column
+
+Sometimes different categories of products have completely different specs, so we don't want to create a column for each of them.
+We could use a JSON column to store the product details. Let's do this by adding a `details` field to our table:
+
+```jsonc
+// POST https://{workspace}.{region}.xata.sh/db/{db}:{branch}/tables/{table}/columns
+
+{
+  "name": "details",
+  "type": "json"
+}
+```
+
+Let's add a few products with different details, for instance:
+
+- A t-shirt with size and color
+- A book with author, ISBN and number of pages
+- A climbing rope with a length and a thickness
+
+<TabbedCode tabs={['TypeScript', 'Python', 'SQL', 'JSON']}>
+```ts
+const record1 = await xata.db.Products.create({
+  name: 'Xata xwag T-shirt',
+  details: {
+    color: 'purple',
+    size: 'M',
+  }
+});
+const record2 = await xata.db.Products.create({
+  name: 'Meditations',
+  details: {
+    author: 'Marcus Aurelius',
+    isbn: '978-0140449334',
+    pages: 304
+  }
+});
+const record3 = await xata.db.Products.create({
+  name: 'Long climbing rope',
+  details: {
+    length: 80,
+    thickness: 9.8,
+    color: 'blue',
+  }
+});
+```
+
+```py
+record1 = xata.records().insert("Products", {
+  "name": "Xata xwag T-shirt",
+  "details": {
+    "color": "purple",
+    "size": "M",
+  }
+})
+record2 = xata.records().insert("Products", {
+  "name": "Meditations",
+  "details": {
+    "author": "Marcus Aurelius",
+    "isbn": "978-0140449334",
+    "pages": 304
+  }
+})
+record3 = xata.records().insert("Products", {
+  "name": "Long climbing rope",
+  "details": {
+    "length": 80,
+    "thickness": 9.8,
+    "color": "blue"
+  }
+})
+```
+
+```jsonc
+    // Not yet available
+```
+
+```jsonc
+// POST https://{workspace}.{region}.xata.sh/db/{db}:{branch}/tables/{table}/data
+{
+  "name": "Xata xwag T-shirt",
+  "details": "{\"color\": \"purple\", \"size\": \"M\"}"
+}
+
+{
+  "name": "Meditations",
+  "details": "{\"author\": \"Marcus Aurelius\", \"isbn\": \"978-0140449334\", \"pages\": 304}"
+}
+
+{
+  "name":"Long climbing rope",
+  "details":"{\"length\": 80, \"thickness\": 9.8, \"color\": \"blue\"}"
+}
+```
+
+</TabbedCode>
+
+It's important to note that the JSON documents are processed and stored in a binary format in order to improve querying and storage performance.
+This has the following implications:
+
+- White spaces are not preserved
+- Key order is not preserved
+- In case of duplicate key, only the last one is stored
+
+## Querying JSON documents
+
```

Comment:
It would be good to have a paragraph here explaining why the `->` syntax. It's ok to say that we use the Postgres `jsonb` and it uses the same syntax as `jsonb`, except that you don't need to cast to text via `->>` (you don't need to, right?)

Reply:
I added a short description of the arrow operator, let me know what you think 🙏

Reply:
LGTM, thanks for adding it!